src/main/java/com/opinta/controller/DiscountPerCounterpartyController.java

```diff
@@ -0,0 +1,114 @@
+    public DiscountPerCounterpartyController(DiscountPerCounterpartyService discountPerCounterpartyService,
+                                             UserService userService) {
+        this.discountPerCounterpartyService = discountPerCounterpartyService;
+        this.userService = userService;
+    }
+    
+    @GetMapping
+    public ResponseEntity<?> getAllDiscountsPerCounterparty(@RequestParam UUID token) {
+        try {
+            User user = userService.authenticate(token);
+            return new ResponseEntity<>(discountPerCounterpartyService.getAll(user), OK);
+        } catch (AuthException e) {
+            return new ResponseEntity<>(getAllOnErrorLogEndpoint(DiscountPerCounterparty.class, e), UNAUTHORIZED);
+        }
+    }
+    
+    @GetMapping("{uuid}")
+    public ResponseEntity<?> getDiscountPerCounterpartyByUuid(@PathVariable UUID uuid, @RequestParam UUID token) {
+        try {
+            User user = userService.authenticate(token);
+            return new ResponseEntity<>(discountPerCounterpartyService.getByUuid(uuid, user), OK);
+        } catch (AuthException e) {
+            return new ResponseEntity<>(authorizationOnErrorLogEndpoint(token, e), UNAUTHORIZED);
+        } catch (IncorrectInputDataException e) {
+            return new ResponseEntity<>(getByIdOnErrorLogEndpoint(DiscountPerCounterparty.class, e), NOT_FOUND);
+        }
+    }
+    
+    @PostMapping
+    public ResponseEntity<?> createDiscountPerCounterparty(
+            @RequestBody DiscountPerCounterpartyDto discountPerCounterpartyDto, @RequestParam UUID token) {
+        try {
+            User user = userService.authenticate(token);
+            return new ResponseEntity<>(discountPerCounterpartyService.save(discountPerCounterpartyDto, user), OK);
+        } catch (AuthException e) {
+            return new ResponseEntity<>(authorizationOnErrorLogEndpoint(token, e), UNAUTHORIZED);
+        } catch (IncorrectInputDataException e) {
+            return new ResponseEntity<>(saveOnErrorLogEndpoint(DiscountPerCounterparty.class, e), NOT_FOUND);
+        } catch (PerformProcessFailedException e) {
+            return new ResponseEntity<>(saveOnErrorLogEndpoint(DiscountPerCounterparty.class, e), BAD_REQUEST);
+        }
+    }
+    
+    @PutMapping("{uuid}")
+    public ResponseEntity<?> updateDiscountPerCounterparty(@PathVariable UUID uuid,
+            @RequestBody DiscountPerCounterpartyDto discountPerCounterpartyDto, @RequestParam UUID token) {
+        try {
+            User user = userService.authenticate(token);
+            return new ResponseEntity<>(discountPerCounterpartyService.update(uuid, discountPerCounterpartyDto, user), OK);
+        } catch (AuthException e) {
+            return new ResponseEntity<>(authorizationOnErrorLogEndpoint(token, e), UNAUTHORIZED);
+        } catch (IncorrectInputDataException e) {
+            return new ResponseEntity<>(updateOnErrorLogEndpoint(DiscountPerCounterparty.class, e), NOT_FOUND);
```

Comment:
Plz, change to updateOnErrorLogEndpoint(DiscountPerCounterparty.class, discountPerCounterpartyDto, e)